bin/brew

```diff
@@ -68,12 +68,55 @@ then
   then
     HOMEBREW_PREFIX="/usr/local"
   fi
+  unset USR_LOCAL_BREW_FILE_DIRECTORY USR_LOCAL_HOMEBREW_REPOSITORY
 fi
 
+unset BREW_FILE_DIRECTORY
+
 # If the location of HOMEBREW_LIBRARY changes
 # keg_relocate.rb, formula_cellar_checks.rb, and test/global_spec.rb need to change.
 HOMEBREW_LIBRARY="${HOMEBREW_REPOSITORY}/Library"
 
+# Load Homebrew's variable configuration files from disk.
+# First, load the system-wide configuration.
+if [[ -f "/etc/homebrew/brew.env" ]]
+then
+  unset SYSTEM_ENV_TAKES_PRIORITY
+  # only load HOMEBREW_*=* lines
+  SYSTEM_HOMEBREW_ENV="$(grep -E '^HOMEBREW_[A-Z]+=.*$' "/etc/homebrew/brew.env")"
+  if [[ -n "${HOMEBREW_SYSTEM_ENV_TAKES_PRIORITY-}" ]]
```

Comment:
Would be nice to check for whether this is set in `/etc/homebrew/brew.env`.

Reply:
@carlocab to avoid it being set by the environment or something else? Allowing it to be set from environment makes testing a bunch easier.

Reply:
More to allow the administrator of `/etc/homebrew/brew.env` to decide whether the system env takes priority or not. Otherwise users can just do `unset HOMEBREW_SYSTEM_ENV_TAKES_PRIORITY`.

But I'm not bothered if you'd rather someone who needed it implemented this.